juju/model.py

```diff
@@ -1177,11 +1224,15 @@ def name(self):
         return self._info.name
 
     @property
-    def info(self):
+    def info(self) -> ModelInfo:
         """Return the cached client.ModelInfo object for this Model.
 
         If Model.get_info() has not been called, this will return None.
```

Comment:
I'm happy to assume that raising a `JujuModelError` instead of returning `None` when `self._info` hasn't been populated is a safe change to make, but let's update the docstring to be accurate.
```suggestion
        Raises a JujuModelError if the model is not connected.
```

Reply:
done